The TP instruction is not supported "~"

### Please describe the bug

For example,/tp 111 111 111 can be executed, but /tp 111~111 cannot be executed

### Steps to reproduce the behavior

As mentioned above, it does not support "~"

### Expected behavior

_No response_

### Server information:

Minecraft Version: 1.20.6
Server Implementation: luminol(Folia Fork) 
Essence Version: 1.6.0

### Please enter a link to your logs

[11:58:04 INFO]: bilibili_Ren issued server command: /tp 111 111 111
[11:58:31 INFO]: bilibili_Ren issued server command: /tp 111 ~ 111
[11:58:31 ERROR]: Command exception: /tp 111 ~ 111
org.bukkit.command.CommandException: Unhandled exception executing command 'tp' in plugin Essence v1.6.0
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[luminol-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:91) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.2.9.jar:?]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:31) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:103) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:448) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.Commands.performCommand(Commands.java:355) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.Commands.performCommand(Commands.java:342) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.commands.Commands.performCommand(Commands.java:337) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2258) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$14(ServerGamePacketListenerImpl.java:2232) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$tryHandleChat$17(ServerGamePacketListenerImpl.java:2396) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at io.papermc.paper.threadedregions.EntityScheduler.executeTick(EntityScheduler.java:173) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1691) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:408) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525) ~[luminol-1.20.6.jar:1.20.6-DEV-e6027bd]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NumberFormatException: For input string: "~"
        at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054) ~[?:?]
        at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110) ~[?:?]
        at java.base/java.lang.Double.parseDouble(Double.java:792) ~[?:?]
        at [E][基础][替代ess的插件]essence-1.6.0.jar/net.lewmc.essence.commands.teleportation.tp.TeleportCommand.onCommand(TeleportCommand.java:64) ~[[E][基础][替代ess的插件]essence-1.6.0.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[luminol-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        ... 20 more

### Anything else?

_No response_

--- a/src/main/java/net/lewmc/essence/commands/teleportation/tp/TeleportCommand.java
+++ b/src/main/java/net/lewmc/essence/commands/teleportation/tp/TeleportCommand.java
@@ -10,6 +10,8 @@ import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Objects;
+
 public class TeleportCommand implements CommandExecutor {
     private final Essence plugin;
     private final LogUtil log;
@@ -71,9 +73,37 @@ public class TeleportCommand implements CommandExecutor {
                         return true;
                     }
 
-                    double x = Double.parseDouble(args[0]);
-                    double y = Double.parseDouble(args[1]);
-                    double z = Double.parseDouble(args[2]);
+                    double x;
+                    double y;
+                    double z;
+
+                    try {
+                        assert player != null;
+                        if (Objects.equals(args[0], "~")) {
+                            x = player.getLocation().getX();
+                        } else {
+                            x = Double.parseDouble(args[0]);
+                        }
+
+                        if (Objects.equals(args[1], "~")) {
+                            y = player.getLocation().getY();
+                        } else {
+                            y = Double.parseDouble(args[1]);
+                        }
+
+                        if (Objects.equals(args[2], "~")) {
+                            z = player.getLocation().getZ();
+                        } else {
+                            z = Double.parseDouble(args[2]);
+                        }
+                    } catch (NullPointerException e) {
+                        message.send("generic", "exception");
+                        this.log.severe(e.getMessage());
+                        return true;
+                    } catch (NumberFormatException e) {
+                        message.send("generic", "numberformaterror");
+                        return true;
+                    }
                     Location location = new Location(player.getWorld(), x, y, z);
                     TeleportUtil tp = new TeleportUtil(this.plugin);
                     tp.doTeleport(player, location, 0);
@@ -94,9 +124,38 @@ public class TeleportCommand implements CommandExecutor {
                             return true;
                         }
 
-                        double x = Double.parseDouble(args[1]);
-                        double y = Double.parseDouble(args[2]);
-                        double z = Double.parseDouble(args[3]);
+                        double x;
+                        double y;
+                        double z;
+
+                        try {
+                            assert player != null;
+                            if (Objects.equals(args[1], "~")) {
+                                x = player.getLocation().getX();
+                            } else {
+                                x = Double.parseDouble(args[1]);
+                            }
+
+                            if (Objects.equals(args[2], "~")) {
+                                y = player.getLocation().getY();
+                            } else {
+                                y = Double.parseDouble(args[2]);
+                            }
+
+                            if (Objects.equals(args[3], "~")) {
+                                z = player.getLocation().getZ();
+                            } else {
+                                z = Double.parseDouble(args[3]);
+                            }
+                        } catch (NullPointerException e) {
+                            message.send("generic", "exception");
+                            this.log.severe(e.getMessage());
+                            return true;
+                        } catch (NumberFormatException e) {
+                            message.send("generic", "numberformaterror");
+                            return true;
+                        }
+
                         Location location = new Location(player.getWorld(), x, y, z);
                         player.teleport(location);
                     } else {
@@ -109,9 +168,37 @@ public class TeleportCommand implements CommandExecutor {
                         LocationUtil locationUtil = new LocationUtil(this.plugin);
                         locationUtil.UpdateLastLocation(playerToTeleport);
 
-                        double x = Double.parseDouble(args[1]);
-                        double y = Double.parseDouble(args[2]);
-                        double z = Double.parseDouble(args[3]);
+                        double x;
+                        double y;
+                        double z;
+
+                        try {
+                            if (Objects.equals(args[1], "~")) {
+                                x = playerToTeleport.getLocation().getX();
+                            } else {
+                                x = Double.parseDouble(args[1]);
+                            }
+
+                            if (Objects.equals(args[2], "~")) {
+                                y = playerToTeleport.getLocation().getY();
+                            } else {
+                                y = Double.parseDouble(args[2]);
+                            }
+
+                            if (Objects.equals(args[3], "~")) {
+                                z = playerToTeleport.getLocation().getZ();
+                            } else {
+                                z = Double.parseDouble(args[3]);
+                            }
+                        } catch (NullPointerException e) {
+                            message.send("generic", "exception");
+                            this.log.severe(e.getMessage());
+                            return true;
+                        } catch (NumberFormatException e) {
+                            message.send("generic", "numberformaterror");
+                            return true;
+                        }
+
                         Location location = new Location(playerToTeleport.getWorld(), x, y, z);
                         TeleportUtil tp = new TeleportUtil(this.plugin);
                         tp.doTeleport(playerToTeleport, location, 0);

--- a/src/main/resources/language/en-GB.yml
+++ b/src/main/resources/language/en-GB.yml
@@ -28,6 +28,7 @@ generic:
   nofolia: "§4[Essence] §cThis command cannot run on Folia-based server software."
   helpfolia: "§4[Essence] §cIf you'd like to help us add support for Folia please visit github.com/lewmc/essence"
   reload: "§2[Essence] §aReloaded."
+  numberformaterror: "§4[Essence] §cEssence encountered an issue whilst converting the format of a number. Please check the command you sent is correct and try again."
 other:
   respawnset: "§6[Essence] §eRespawn point set."
   rule: "§a{{1}}"

